pymc/distributions/discrete.py

```diff
@@ -1224,7 +1226,16 @@ class Categorical(Discrete):
 
     @classmethod
     def dist(cls, p, **kwargs):
-
+        if isinstance(p, np.ndarray) or isinstance(p, list):
+            if (np.array(p) < 0).any():
+                raise ValueError("Negative probabilities are not valid")
+            p_sum = np.sum([p], axis=-1)
+            if (p_sum != 1.0).any(): 
```

Comment:
We probably have to use something more soft due to floating point errors:

```suggestion
            if not np.all(np.isclose(p_sum, 1.0)): 
```